ppsci/arch/kan.py

```diff
@@ -0,0 +1,383 @@
+# Copyright (c) 2025 PaddlePaddle Authors. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from typing import Tuple
+
+import paddle
+
+from ppsci.arch import base
+from ppsci.utils import initializer
+
+"""
+This is the paddle implementation of Korogonov-Arnold-Network (KAN)
+which is based on the torch implementation [efficient-kan] by Blealtan and akkashdash
+please refer to their work (https://github.com/Blealtan/efficient-kan)
+Authors: guhaohao0991(guhaohao@baidu.com)
+Date:    2025/04/
+"""
+
+
+class KANLinear(paddle.nn.Layer):
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int,
+        grid_size: int = 5,
+        spline_order: int = 3,
+        scale_noise=0.1,
+        scale_base=1.0,
+        scale_spline=1.0,
+        enable_standalone_scale_spline: bool = True,
+        base_activation=paddle.nn.Silu,
+        grid_eps=0.02,
+        grid_range=[-1, 1],
```

Comment:
参数尽量不要用list这种可变类型，改为tuple
```suggestion
        grid_range=(-1, 1),
```